### PR TITLE
Move to ex_doc -style documentation

### LINKED ISSUE #121: docs: move to `ex_doc`
It'd be nice to review `rebar.config` and then move to `rebar ex_doc` as a way to generate more recent docs (also, as supported natively by OTP 27).

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
**Note**: Depends on the CI file updates from https://github.com/elli-lib/elli/pull/114, so shall be rebased and updated (if required) after that one. Closes #121.

## What this pull request is

* moving from a EDoc/.md -style to an ExDoc/Hexdocs.pm -style, plus
* making sure `rebar3_ex_doc` runs smoothly on this, via `rebar3 ex_doc`, thus
* modernising the documentation

## What this pull request is not

* an exhaustive attempt at documenting Elli's internals (or even changing the user-facing pages significantly)

## Bonus

* .md files were linted and adjusted according to results

## Notes

* I completely removed the `doc` folder, since I expect the documentation to be hosted next to hexdocs.pm, but lemme know if this was a good call
* I call `elli` to the modules, but Elli to the project consistently (this wasn't 100% done)

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
I did this change without first checking if the Issues or Pull requests mentioned it (glad they didn't, so my time wasn't wasted 😄), but now see it might even fit well with #102, that mentions:

* #38 (I think we just need to make sure the docs get published correctly)

**Comment by paulo-ferraz-oliveira** [comment]
CI failing for OTP 26. I'll try to replicate locally.

**Comment by paulo-ferraz-oliveira** [comment]
Removed OTP 26 from scope, since there're SSL -related issues to solve. Should be moved to a new PR, I guess...

**Review by tsloughter — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
Is anything lacking for this one to get merged? I can then rebase and update the other ones.

**Comment by paulo-ferraz-oliveira** [comment]
Moved to draft. These don't need to be reviewed before #114, after which I'll rebase.

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter

**Comment by tsloughter** [comment]
My only concern is people likely got used to having the docs there in the github repo. But moving to exdoc is def needed and the future, seeing as OTP itself has done so. Perfect world we could have both.

Resolve the conflicts and I'll merge.

**Comment by paulo-ferraz-oliveira** [comment]
I don't mind pushing the `docs` to the repo, if that makes sense. I'll test it in this pull request, since I don't know how it'd be rendered/accessible.

Edit: I don't believe I've removed it, so am wondering **where** "in the github repo" you could see this? Rendered on the _source code_ directly?

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, tests are passing. Lemme know if you want me to change something, or we can also create issues and look at it later, if it's not blocking. My goal would be, after this, to get tests running on 26 and 27, and then request a release.

**Comment by tsloughter** [comment]
@paulo-ferraz-oliveira sorry, wasn't clear. I meant just if it could be markdown. Since it can't be generated to markdown we can remove it. After you remove it I'll merge this.

**Comment by paulo-ferraz-oliveira** [comment]
> I meant just if it could be markdown.

What I thought 😄 Alas, it can't. But should it be possible? I mean, we write Markdown in the doc.s, then it gets converted to HTML, so surely something can be done (?)

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, I think now we're ready 😄

**Review by tsloughter — APPROVED** [review]
Hope everyone is good with this move and the removal of the markdown :). I think it is the right call, esp given OTP has even moved to ex_doc.

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, shall we tag somebody to do extra validation on this, or is it Ok?

**Comment by paulo-ferraz-oliveira** [comment]
@tsloughter, shall we tag somebody to do extra validation on this, or is it Ok?

File: src/elli_request.erl
Comment (yurrriq) [inline src/elli_request.erl:178]: Will ex_doc make a link here?

File: src/elli_http.erl
Comment (yurrriq) [inline src/elli_http.erl]: Is there some particular reason to drop the `` ` `` and `` ' `` here?

**Review by yurrriq — APPROVED** [review]
lgtm, just left a couple comments

File: src/elli_request.erl
Comment (paulo-ferraz-oliveira) [reply src/elli_request.erl:178]: It should, yeah; it's pretty good at that. But lemme try it out, locally.

Comment (paulo-ferraz-oliveira) [reply src/elli_request.erl:178]: image

File: src/elli_http.erl
Comment (paulo-ferraz-oliveira) [reply src/elli_http.erl]: No special reason. Lemme re-check the links.

Comment (paulo-ferraz-oliveira) [reply src/elli_http.erl]: Most likely because it's a link to an external project, so it wouldn't know where to point, but I can check...

Comment (paulo-ferraz-oliveira) [reply src/elli_http.erl]: Unless you mean you want to keep the notation even if there's no link, which seems to be the case.

**Comment by paulo-ferraz-oliveira** [comment]
Push two minor changes (latest commits) based on review and self-review. I believe it's possible to publish the docs alone (to test them out?) even though I've never done it before.

**Comment by paulo-ferraz-oliveira** [comment]
This has 2 approvals but not merged; I can't merge either, since I don't maintain the project 😄. Is there something missing, still?

**Comment by tsloughter** [comment]
Sorry, no, just forgot. Thanks!

Next up, move to `-doc()` ;)

**Pull request merged**
